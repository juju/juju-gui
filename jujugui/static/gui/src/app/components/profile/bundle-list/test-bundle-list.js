/* Copyright (C) 2017 Canonical Ltd. */

'use strict';
const React = require('react');

const ProfileBundleList = require('./bundle-list');

const jsTestUtils = require('../../../utils/component-test-utils');

describe('Profile Bundle List', function() {

  const rawBundleData = `[{
    "id": "cs:~lazypower/bundle/logstash-core-1",
    "perm": {
      "read": ["everyone"],
      "write": ["lazypower"]
    },
    "applications": {
      "elasticsearch": {
        "charm": "cs:~lazypower/trusty/elasticsearch"
      },
      "kibana": {
        "charm": "cs:trusty/kibana-10"
      },
      "logstash": {
        "charm": "cs:~lazypower/trusty/logstash-20"
      },
      "openjdk": {
        "charm": "cs:~kwmonroe/trusty/openjdk"
      }
    },
    "name": "logstash-core",
    "unitCount": 3
  }, {
    "id": "cs:~lazypower/bundle/swarm-core-1",
    "perm": {
      "read": ["lazypower", "everyone"],
      "write": ["lazypower"]
    },
    "applications": {
      "consul": {
        "charm": "cs:~containers/trusty/consul"
      },
      "swarm": {
        "charm": "cs:~lazypower/swarm"
      }
    },
    "name": "swarm-core",
    "unitCount": 5
  }]`;

  function renderComponent(options={}) {
    const charmstoreList = (user, cb) => {
      assert.equal(user, 'lazypower@external');
      cb(null, JSON.parse(rawBundleData));
    };
    return jsTestUtils.shallowRender(
      <ProfileBundleList
        addNotification={sinon.stub()}
        baseURL="/gui/"
        changeState={options.changeState || sinon.stub()}
        charmstore={{
          list: charmstoreList,
          url: '/charmstore'
        }}
        user="lazypower@external" />, true);
  }

  it('can render', () => {
    const renderer = renderComponent();
    const output = renderer.getRenderOutput();
    const list = output.props.children.props.children[1];
    const expected = (
      <div className="profile-bundle-list">
        <ul>
          <li className="profile-bundle-list__table-header">
            <span>Name</span>
            <span>Units</span>
            <span>Owner</span>
            <span>Visibility</span>
          </li>
          <li className="profile-bundle-list__row">
            <span>
              <img
                className="profile-bundle-list__icon"
                src="/charmstore/~lazypower/trusty/elasticsearch/icon.svg"
                title="logstash-core"/>
              <a
                href="/gui/u/lazypower/logstash-core/bundle/1"
                onClick={list[0].props.children[0].props.children[1].props.onClick}>
                logstash-core
              </a>
            </span>
            <span>3</span>
            <span>lazypower@external</span>
            <span>public</span>
          </li>
          <li className="profile-bundle-list__row">
            <span>
              <img
                className="profile-bundle-list__icon"
                src="/charmstore/~containers/trusty/consul/icon.svg"
                title="swarm-core"/>
              <a
                href="/gui/u/lazypower/swarm-core/bundle/1"
                onClick={list[1].props.children[0].props.children[1].props.onClick}>
                swarm-core
              </a>
            </span>
            <span>5</span>
            <span>lazypower@external</span>
            <span>public</span>
          </li>
        </ul>
      </div>);
    expect(output).toEqualJSX(expected);
  });
});
