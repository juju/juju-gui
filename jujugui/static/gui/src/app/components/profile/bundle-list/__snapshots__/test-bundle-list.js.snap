// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Bundle List can render 1`] = `
<div
  className="profile-bundle-list"
>
  <div>
    <div
      className="v1"
    >
      <h2
        className="profile__title"
      >
        My
         bundles
        <span
          className="profile__title-count"
        >
          (
          2
          )
        </span>
      </h2>
    </div>
    <BasicTable
      headerClasses={
        Array [
          "profile__entity-table-header-row",
        ]
      }
      headerColumnClasses={
        Array [
          "profile__entity-table-header-column",
        ]
      }
      headers={
        Array [
          Object {
            "columnSize": 6,
            "content": "Name",
          },
          Object {
            "classes": Array [
              "u-align--right",
            ],
            "columnSize": 2,
            "content": "Machines",
          },
          Object {
            "classes": Array [
              "u-align--right",
            ],
            "columnSize": 1,
            "content": "Units",
          },
          Object {
            "columnSize": 3,
            "content": "Release",
          },
        ]
      }
      rowClasses={
        Array [
          "profile__entity-table-row",
        ]
      }
      rowColumnClasses={
        Array [
          "profile__entity-table-column",
        ]
      }
      rows={
        Array [
          Object {
            "columns": Array [
              Object {
                "columnSize": 3,
                "content": <a
                  href="/gui/u/lazypower/logstash-core/bundle/1"
                  onClick={[Function]}
                >
                  logstash-core
                </a>,
              },
              Object {
                "columnSize": 3,
                "content": <IconList
                  applications={
                    Array [
                      Object {
                        "displayName": "elasticsearch",
                        "iconPath": "/charmstore/~lazypower/trusty/elasticsearch/icon.svg",
                        "id": "cs:~lazypower/trusty/elasticsearch",
                      },
                      Object {
                        "displayName": "kibana",
                        "iconPath": "/charmstore/trusty/kibana-10/icon.svg",
                        "id": "cs:trusty/kibana-10",
                      },
                      Object {
                        "displayName": "logstash",
                        "iconPath": "/charmstore/~lazypower/trusty/logstash-20/icon.svg",
                        "id": "cs:~lazypower/trusty/logstash-20",
                      },
                      Object {
                        "displayName": "openjdk",
                        "iconPath": "/charmstore/~kwmonroe/trusty/openjdk/icon.svg",
                        "id": "cs:~kwmonroe/trusty/openjdk",
                      },
                    ]
                  }
                  changeState={[Function]}
                  generatePath={[Function]}
                />,
              },
              Object {
                "classes": Array [
                  "u-align--right",
                ],
                "columnSize": 2,
                "content": 2,
              },
              Object {
                "classes": Array [
                  "u-align--right",
                ],
                "columnSize": 1,
                "content": 3,
              },
              Object {
                "columnSize": 3,
                "content": "#1",
              },
            ],
            "expandedContent": <ProfileExpandedContent
              acl={
                Object {
                  "isReadOnly": [Function],
                }
              }
              addToModel={[Function]}
              changeState={[Function]}
              entity={
                Object {
                  "applications": Object {
                    "elasticsearch": Object {
                      "charm": "cs:~lazypower/trusty/elasticsearch",
                    },
                    "kibana": Object {
                      "charm": "cs:trusty/kibana-10",
                    },
                    "logstash": Object {
                      "charm": "cs:~lazypower/trusty/logstash-20",
                    },
                    "openjdk": Object {
                      "charm": "cs:~kwmonroe/trusty/openjdk",
                    },
                  },
                  "bugUrl": "example.com/bugs",
                  "description": "logstash-core description",
                  "homepage": "example.com/",
                  "id": "cs:~lazypower/bundle/logstash-core-1",
                  "machineCount": 2,
                  "name": "logstash-core",
                  "perm": Object {
                    "read": Array [
                      "everyone",
                    ],
                    "write": Array [
                      "lazypower",
                    ],
                  },
                  "unitCount": 3,
                }
              }
              generatePath={[Function]}
              getDiagramURL={[Function]}
              getModelName={[Function]}
              topRow={
                <div>
                  <div
                    className="six-col profile-expanded-content__top-row"
                  >
                    <a
                      href="/gui/u/lazypower/logstash-core/bundle/1"
                      onClick={[Function]}
                    >
                      logstash-core
                    </a>
                  </div>
                  <div
                    className="two-col profile-expanded-content__top-row u-align--right"
                  >
                    2
                  </div>
                  <div
                    className="one-col profile-expanded-content__top-row u-align--right"
                  >
                    3
                  </div>
                  <div
                    className="three-col last-col profile-expanded-content__top-row"
                  >
                    #1
                  </div>
                </div>
              }
            />,
            "extraData": "logstash-core",
            "key": "cs:~lazypower/bundle/logstash-core-1",
          },
          Object {
            "columns": Array [
              Object {
                "columnSize": 3,
                "content": <a
                  href="/gui/u/lazypower/swarm-core/bundle/1"
                  onClick={[Function]}
                >
                  swarm-core
                </a>,
              },
              Object {
                "columnSize": 3,
                "content": <IconList
                  applications={
                    Array [
                      Object {
                        "displayName": "consul",
                        "iconPath": "/charmstore/~containers/trusty/consul/icon.svg",
                        "id": "cs:~containers/trusty/consul",
                      },
                      Object {
                        "displayName": "swarm",
                        "iconPath": "/charmstore/~lazypower/swarm/icon.svg",
                        "id": "cs:~lazypower/swarm",
                      },
                    ]
                  }
                  changeState={[Function]}
                  generatePath={[Function]}
                />,
              },
              Object {
                "classes": Array [
                  "u-align--right",
                ],
                "columnSize": 2,
                "content": 5,
              },
              Object {
                "classes": Array [
                  "u-align--right",
                ],
                "columnSize": 1,
                "content": 5,
              },
              Object {
                "columnSize": 3,
                "content": "#1",
              },
            ],
            "expandedContent": <ProfileExpandedContent
              acl={
                Object {
                  "isReadOnly": [Function],
                }
              }
              addToModel={[Function]}
              changeState={[Function]}
              entity={
                Object {
                  "applications": Object {
                    "consul": Object {
                      "charm": "cs:~containers/trusty/consul",
                    },
                    "swarm": Object {
                      "charm": "cs:~lazypower/swarm",
                    },
                  },
                  "bugUrl": "example.com/bugs",
                  "description": "swarm-core description",
                  "homepage": "example.com/",
                  "id": "cs:~lazypower/bundle/swarm-core-1",
                  "machineCount": 5,
                  "name": "swarm-core",
                  "perm": Object {
                    "read": Array [
                      "lazypower",
                      "everyone",
                    ],
                    "write": Array [
                      "lazypower",
                    ],
                  },
                  "unitCount": 5,
                }
              }
              generatePath={[Function]}
              getDiagramURL={[Function]}
              getModelName={[Function]}
              topRow={
                <div>
                  <div
                    className="six-col profile-expanded-content__top-row"
                  >
                    <a
                      href="/gui/u/lazypower/swarm-core/bundle/1"
                      onClick={[Function]}
                    >
                      swarm-core
                    </a>
                  </div>
                  <div
                    className="two-col profile-expanded-content__top-row u-align--right"
                  >
                    5
                  </div>
                  <div
                    className="one-col profile-expanded-content__top-row u-align--right"
                  >
                    5
                  </div>
                  <div
                    className="three-col last-col profile-expanded-content__top-row"
                  >
                    #1
                  </div>
                </div>
              }
            />,
            "extraData": "swarm-core",
            "key": "cs:~lazypower/bundle/swarm-core-1",
          },
        ]
      }
      sort={[Function]}
    />
  </div>
</div>
`;
